crates/bevy_render/src/mesh/primitives/circle_iterator.rs

```diff
@@ -0,0 +1,80 @@
+use bevy_math::Vec2;
+use std::f32::consts::TAU;
+
+#[derive(Debug, Clone)]
+pub(crate) struct CircleIterator {
+    count: usize,
+    theta: f32,
+    wrap: bool,
+}
+
+impl CircleIterator {
+    pub(crate) fn new(count: usize, wrap: bool) -> CircleIterator {
+        if wrap {
+            Self {
+                count: count + 1,
+                theta: -TAU / (count as f32),
+                wrap,
+            }
+        } else {
+            Self {
+                count,
+                theta: -TAU / (count as f32),
+                wrap,
+            }
+        }
+    }
+}
+impl Iterator for CircleIterator {
+    type Item = Vec2;
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.count != 0 {
+            if self.wrap {
+                self.count -= 1;
+                Some(Vec2::new(
+                    (self.theta * self.count as f32).cos(),
```

Comment:
Why not `sin_cos`?

Reply:
The main goal of this step was to duplicate the existing logic and relocate it to a unified function, and then validate that it hasn't impacted performance or broken the existing meshing. I do agree that sin_cos would make sense, though.